### ISSUE TITLE
Ignoring tabs warning causes incorrect indentation warning

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#453]
In GitLab by @darricktheprogrammer on Nov 3, 2015, 13:59

If you ignore warning `W191: indentation contains tabs` and line up brackets/parentheses with the first non-whitespace character of the last line instead of the first character of the opening line, flake8 returns error `E123 closing bracket does not match indentation of opening bracket's line`.

This does not receive an error:

```python
d = {
    'a': 'Axxx', 'b': 'Bxxx', 'c': 'Cxxx', 'd': 'Dxxx', 'e': 'Exxx',
    'f': 'Fxxx', 'g': 'Gxxx', 'h': 'Hxxx', 'i': 'Ixxx', 'j': 'Jxxx',
    'k': 'Kxxx', 'l': 'Lxxx', 'm': 'Mxxx', 'n': 'Nxxx', 'o': 'Oxxx',
    'p': 'Pxxx', 'q': 'Qxxx', 'r': 'Rxxx', 's': 'Sxxx', 't': 'Txxx',
    'u': 'Uxxx', 'v': 'Vxxx', 'w': 'Wxxx', 'x': 'Xxxx', 'y': 'Yxxx',
    'z': 'Zxxx',
}
```

This receives an error:

```python
d = {
    'a': 'Axxx', 'b': 'Bxxx', 'c': 'Cxxx', 'd': 'Dxxx', 'e': 'Exxx',
    'f': 'Fxxx', 'g': 'Gxxx', 'h': 'Hxxx', 'i': 'Ixxx', 'j': 'Jxxx',
    'k': 'Kxxx', 'l': 'Lxxx', 'm': 'Mxxx', 'n': 'Nxxx', 'o': 'Oxxx',
    'p': 'Pxxx', 'q': 'Qxxx', 'r': 'Rxxx', 's': 'Sxxx', 't': 'Txxx',
    'u': 'Uxxx', 'v': 'Vxxx', 'w': 'Wxxx', 'x': 'Xxxx', 'y': 'Yxxx',
    'z': 'Zxxx',
    }
```

Here is the output, run back to back on the same file, showing that output is fine, unless the option to ignore `W191` is flagged.

```
darrick@computername [~]: flake8 ~/Desktop/indentation.py
darrick@computername [~]: flake8 --ignore=W191 ~/Desktop/indentation.py
/Users/darrick/Desktop/indentation.py:8:5: E123 closing bracket does not match indentation of opening bracket's line
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 3, 2015, 14:33

Can you provide the output of `flake8 --version`.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 3, 2015, 14:35

Oh I see what the problem is. Using `--ignore` does not append to the list. So the default ignore list is overridden by what you pass. You can do the following:

```
flake8 --ignore=E121,E123,E126,E226,E24,E704,W191 ~/Desktop/indentation.py
```

**Comment by asottile** [comment]
In GitLab by @darricktheprogrammer on Nov 3, 2015, 15:31

Version:

    2.3.0 (pep8: 1.6.2, pyflakes: 0.8.1, mccabe: 0.3) CPython 2.7.6 on Darwin

I also tried v2.5, hoping it might be fixed, but the same thing happens in that version.

I realize I can ignore the `E123` error, but I still want to get the warnings if the close bracket is not in one of the two locations laid out in pep8.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 3, 2015, 20:38

If you want `E123` always on then you need to specify a separate ignore list than the default. For example,

```
flake8 --ignore=E121,E126,E226,E24,E704 ...
```

pep8 has a list of errors that are ignored by default because either PEP-0008 changed and those rules are now invalid, or because they overstepped the document in the first place and were controversial. You can turn them back on by specifying a different list of things you'd like to ignore.

Can you maybe tell me exactly what you're expecting to happen and what exactly you think is the bug?

**Comment by asottile** [comment]
In GitLab by @darricktheprogrammer on Nov 4, 2015, 05:24

I see what happened now. I have a lot of whitespace errors ignored in my config file because pep8 doesn't always read tabs correctly (I use tabs instead not spaces), but I didn't add those to the ignore list in my simplified version for the bug report. I did not realize that pep8 has a default ignore list that is being turned off once you add anything to `--ignore`.
